pallets/connectors-gateway/src/tests.rs

```diff
@@ -0,0 +1,480 @@
+
+	use super::*;
+
+	#[test]
+	fn success() {
+		new_test_ext().execute_with(|| {
+			let address = H160::from_slice(&get_random_test_account_id().as_slice()[..20]);
+			let domain_address = DomainAddress::EVM(0, address.into());
+
+			assert_ok!(ConnectorsGateway::add_connector(
+				RuntimeOrigin::root(),
+				domain_address.clone(),
+			));
+
+			assert!(ConnectorsAllowlist::<Runtime>::contains_key(
+				domain_address.domain(),
+				domain_address.clone()
+			));
+
+			event_exists(Event::<Runtime>::ConnectorAdded {
+				connector: domain_address.clone(),
+			});
+
+			let expected_msg = MessageMock::First;
+			let encoded_msg = expected_msg.serialize();
+
+			let expected_domain_address = domain_address.clone();
+
+			MockConnectors::mock_submit(move |domain, message| {
+				assert_eq!(domain, expected_domain_address);
+				assert_eq!(message, expected_msg);
+				Ok(())
+			});
+
+			assert_ok!(ConnectorsGateway::process_msg(
+				GatewayOrigin::Local(domain_address).into(),
+				BoundedVec::<u8, MaxIncomingMessageSize>::try_from(encoded_msg).unwrap()
+			));
+		});
+	}
+
+	#[test]
+	fn bad_origin() {
+		new_test_ext().execute_with(|| {
+			let encoded_msg = MessageMock::First.serialize();
+
+			assert_noop!(
+				ConnectorsGateway::process_msg(
+					RuntimeOrigin::root(),
+					BoundedVec::<u8, MaxIncomingMessageSize>::try_from(encoded_msg).unwrap()
+				),
+				BadOrigin,
+			);
+		});
+	}
+
+	#[test]
+	fn invalid_message_origin() {
+		new_test_ext().execute_with(|| {
+			let domain_address = DomainAddress::Centrifuge(get_random_test_account_id().into());
+			let encoded_msg = MessageMock::First.serialize();
+
+			assert_noop!(
+				ConnectorsGateway::process_msg(
+					GatewayOrigin::Local(domain_address).into(),
+					BoundedVec::<u8, MaxIncomingMessageSize>::try_from(encoded_msg).unwrap()
+				),
+				Error::<Runtime>::InvalidMessageOrigin,
+			);
+		});
+	}
+
+	#[test]
+	fn unknown_connector() {
+		new_test_ext().execute_with(|| {
+			let address = H160::from_slice(&get_random_test_account_id().as_slice()[..20]);
+			let domain_address = DomainAddress::EVM(0, address.into());
+			let encoded_msg = MessageMock::First.serialize();
+
+			assert_noop!(
+				ConnectorsGateway::process_msg(
+					GatewayOrigin::Local(domain_address).into(),
+					BoundedVec::<u8, MaxIncomingMessageSize>::try_from(encoded_msg).unwrap()
+				),
+				Error::<Runtime>::UnknownConnector,
+			);
+		});
+	}
+
+	#[test]
+	fn message_decode() {
+		new_test_ext().execute_with(|| {
+			let address = H160::from_slice(&get_random_test_account_id().as_slice()[..20]);
+			let domain_address = DomainAddress::EVM(0, address.into());
+
+			assert_ok!(ConnectorsGateway::add_connector(
+				RuntimeOrigin::root(),
+				domain_address.clone(),
+			));
+
+			assert!(ConnectorsAllowlist::<Runtime>::contains_key(
+				domain_address.domain(),
+				domain_address.clone()
+			));
+
+			event_exists(Event::<Runtime>::ConnectorAdded {
+				connector: domain_address.clone(),
+			});
```

Comment:
Nit: See above, no need to repeat these checks if we have properly ensured `add_connector` beforehand as you did. This reduces noise in the tests.